folium/plugins/timedynamic_geo_json.py

```diff
@@ -0,0 +1,175 @@
+
+    """
+    def __init__(self, data, styledict, **kwargs):
+        super(TimeDynamicGeoJson, self).__init__(data, **kwargs)
+        assert isinstance(styledict, dict), 'styledict must be a dictionary'
+        for val in styledict.values():
+            assert isinstance(val, dict), 'each item in styledict must be a dictionary'
+
+        self.styledict = styledict
+
+        # make set of timestamps
+        self.timestamps = set()
+        for feature in self.styledict.values():
+            self.timestamps.update(set(feature.keys()))
+        self.timestamps = sorted(list(self.timestamps))
+
+        self._template = Template(u"""
+            {% macro script(this, kwargs) %}
+
+                var timestamps = {{ this.timestamps }};
+                var styledict = {{ this.styledict }};
+                var current_timestamp = timestamps[0];
+
+                // insert time slider
+                d3.select("body").insert("p", ":first-child").append("input")
+                    .attr("type", "range")
+                    .attr("width", "100px")
+                    .attr("min", 0)
+                    .attr("max", timestamps.length - 1)
+                    .attr("value", 0)
+                    .attr("id", "slider")
+                    .attr("step", "1")
+                    .style('align', 'center');
+
+                // insert time slider output BEFORE time slider (text on top of slider)
+                d3.select("body").insert("p", ":first-child").append("output")
+                    .attr("width", "100")
+                    .attr("id", "slider-value")
+                    .style('font-size', '18px')
+                    .style('text-align', 'center')
+                    .style('font-weight', '500%');
+
+                var datestring = new Date(parseInt(current_timestamp)*1000).toDateString();
+                d3.select("output#slider-value").text(datestring);
+
+                fill_map = function(){
+                    for (var feature_id in styledict){
+                        let style = styledict[feature_id]//[current_timestamp];
+                        var fillColor = 'white';
+                        var opacity = 0;
+                        if (current_timestamp in style){
+                            fillColor = style[current_timestamp]['color'];
+                            opacity = style[current_timestamp]['opacity'];
+                            d3.selectAll('#feature-'+feature_id).attr('fill', fillColor).style('fill-opacity', opacity);
+                        }
+                    }
+                }
+
+                d3.select("#slider").on("input", function() {
+                    current_timestamp = timestamps[this.value];
+                var datestring = new Date(parseInt(current_timestamp)*1000).toDateString();
+                d3.select("output#slider-value").text(datestring);
+                fill_map();
+                });
+
+                {% if this.highlight %}
```

Comment:
Note to self: unfortunately this repetition is needed b/c of the current design we cannot extend the jinja template.